core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMap.kt

```diff
@@ -52,12 +52,21 @@ interface DistanceRangeMap<T> : Iterable<DistanceRangeMap.RangeMapEntry<T>> {
 
     /**
      * Updates the map with another one, using a merge function to fuse the values of intersecting
-     * ranges
+     * ranges. Doesn't keep any range from update where there is no intersection.
      */
-    fun <U> updateMap(update: DistanceRangeMap<U>, updateFunction: BiFunction<T, U, T>)
+    fun <U> updateMapIntersection(update: DistanceRangeMap<U>, updateFunction: BiFunction<T, U, T>)
+
+    /**
+     * Updates the map with another one, using a merge function to fuse the values of intersecting
+     * ranges. Keeps the values of the ranges from update where there is no intersection.
+     */
+    fun updateMap(update: DistanceRangeMap<T>, updateFunction: BiFunction<T, T, T>)
```

Comment:
This would feel more natural to me to have an `updateFunction`  that's always applied:
```suggestion
    fun updateMap(update: DistanceRangeMap<T>, updateFunction: BiFunction<T?, T?, T>)
```
The current way feels like a trap (documented, so it's still OK I guess) if the updateFunction is something like `a^2+b^2`.

Reply:
I took the middle ground where you can provide an additional mapping function for non overlapping ranges.

The usability of implementing `if (a == null) {a} else if (b == null) {b} else{ /*...*/}` is quite low for something I'm not even sure we will use